internal/metrics/listener.go

```diff
@@ -49,5 +53,47 @@ func runServer(port string, registry prometheus.Gatherer) {
 	})
 	address := fmt.Sprintf(":%v", port)
 	glog.Infof("Starting Prometheus listener on: %v%v", address, metricsEndpoint)
-	glog.Fatal("Error in Prometheus listener server: ", http.ListenAndServe(address, nil))
+	if prometheusSecret == nil {
+		glog.Fatal("Error in Prometheus listener server: ", http.ListenAndServe(address, nil))
+	} else {
+		// Unfortunately, http.ListenAndServeTLS() takes a filename instead of cert/key data, so we
+		// Write the cert and key to a temporary file. We create a unique file name to prevent collisions.
+		certFileName := fmt.Sprintf("%v-%s", time.Now().Unix(), "nginx-prometheus.cert")
+		keyFileName := fmt.Sprintf("%v-%s", time.Now().Unix(), "nginx-prometheus.key")
+		certFile, err := writeTempFile(prometheusSecret.Data[api_v1.TLSCertKey], certFileName)
+		if err != nil {
+			glog.Errorf("failed to create cert file for prometheus: %w", err)
+		}
+		defer removeTemporaryFile(certFile)
+
+		keyFile, err := writeTempFile(prometheusSecret.Data[api_v1.TLSPrivateKeyKey], keyFileName)
+		if err != nil {
+			glog.Errorf("failed to create key file for prometheus: %w", err)
+		}
+		defer removeTemporaryFile(keyFile)
+
+		glog.Fatal("Error in Prometheus listener server: ", http.ListenAndServeTLS(address, certFile.Name(), keyFile.Name(), nil))
+	}
+}
+
+func writeTempFile(data []byte, name string) (*os.File, error) {
+	f, err := ioutil.TempFile("", name)
```

Comment:
at this point, the name is already unique, because it includes the date prefix before the `nginx-prometheus.cert` and `nginx-prometheus.key`.   In `ioutil.TempFile("", name)` "the filename is generated by taking pattern and adding a random string to the end". So we basically we're doing the uniqueness work twice. Is it necessary? Do we need to have the date prefix?

Also note that it makes sense to store it with `600` permission so only the process under `nginx` user has access to it - this user is used to run both the IC and NGINX.  This is similar to the permission of the regular TLS secrets the IC writes:
```
	TLSSecretFileMode            = 0600  // TLSSecretFileMode defines the default file mode for files with TLS Secrets.
```